File: Pantry_Label_Frontend/src/services/productService.js
import axios from 'axios';

const API_BASE = import.meta.env.VITE_API_URL || 'http://localhost:8080';

export const productService = {
  getStates: async () => {
    return axios.get(`${API_BASE}/api/states`);
  },

  getCategories: async () => {
    return axios.get(`${API_BASE}/api/categories`);
  },

  getProducts: async (stateId, categoryId = null, searchQuery = '') => {
    const params = new URLSearchParams();
    if (stateId) params.append('state', stateId);
    if (categoryId) params.append('category', categoryId);
    if (searchQuery) params.append('search', searchQuery);

    return axios.get(`${API_BASE}/api/products`, { params });
  }
};

export default productService;

File: Pantry_Label_Backend/src/main/java/com/pantrylabel/service/ProductService.java
package com.pantrylabel.service;
import com.pantrylabel.dto.response.CategoryDto;
import com.pantrylabel.dto.response.ProductDto;
import com.pantrylabel.dto.response.StateDto;
import java.util.List;
public interface ProductService { List<StateDto> getStates(); List<CategoryDto> getCategories(); List<ProductDto> getProducts(Long stateId, Long categoryId, String search); }
